tiny_dnn/core/framework/tensor.h

```diff
@@ -108,7 +110,17 @@ class Tensor {
     const float_t& operator[] (cnn_size_t index) const {
         return *access_data(index);
     }
-    
+
+    void toDevice(const Device& device) {
+#if defined(USE_OPENCL) || defined(USE_CUDA)
+        CLCudaAPI::Context ctx = device.context();
+        CLCudaAPI::Queue queue = device.queue();
+
+        data_gpu_ = std::make_shared<CLCudaAPI::Buffer<float_t> >(
+            ctx, queue, data_cpu_->begin(), data_cpu_->end());
+#endif
+    }
+
```

Comment:
It could be interested to know if @CNugteren will support [RDMA](https://streamcomputing.eu/blog/2015-04-18/gpudirect-and-directgma-direct-gpu-gpu-communication/) (probably we need a more update link) between devices in clcudaapi.